src/integrations/AWSIntegration.ts

```diff
@@ -616,6 +618,84 @@ export class AWSSNSIntegration {
+    public static createSpan(tracer: any, request: any, config: any): ThundraSpan {
+        const operationName = request.operation ? request.operation : AWS_SERVICE_REQUEST;
+        const operationType = AWSIntegration.getOperationType(operationName, ClassNames.STEPFUNCTIONS);
+
+        const spanName = AWSStepFunctionsIntegration.getStateMachineName(request) || AWS_SERVICE_REQUEST;
+        const parentSpan = tracer.getActiveSpan();
+        const activeSpan = tracer._startSpan(spanName, {
+            childOf: parentSpan,
+            domainName: DomainNames.AWS,
+            className: ClassNames.STEPFUNCTIONS,
+            disableActiveStart: true,
+            tags: {
+                [SpanTags.OPERATION_TYPE]: operationType,
+                [SpanTags.SPAN_TYPE]: SpanTypes.AWS_STEPFUNCTIONS,
+                [AwsSDKTags.REQUEST_NAME]: operationName,
+            },
+        });
+
+        AWSStepFunctionsIntegration.createStepFunctionTraceLink(request, activeSpan);
+
+        const stateMachineARN = get(request, 'params.stateMachineArn', '');
+        const executionName = get(request, 'params.name', '');
+
+        activeSpan.setTag(AwsStepFunctionsTags.STATE_MACHINE_ARN, stateMachineARN);
+        activeSpan.setTag(AwsStepFunctionsTags.EXECUTION_NAME, executionName);
+        activeSpan.setTag(SpanTags.TOPOLOGY_VERTEX, true);
+        activeSpan.setTag(SpanTags.TRIGGER_DOMAIN_NAME, LAMBDA_APPLICATION_DOMAIN_NAME);
+        activeSpan.setTag(SpanTags.TRIGGER_CLASS_NAME, LAMBDA_APPLICATION_CLASS_NAME);
+
+        return activeSpan;
+    }
+
+    public static createTraceLinks(span: ThundraSpan, request: any, config: any): any[] {
+        return [];
+    }
+
+    public static processResponse(span: ThundraSpan, request: any, config: any): void {
+        const executionARN = get(request, 'response.data.executionArn', '');
+        const startDate = get(request, 'response.data.startDate');
+
+        span.setTag(AwsStepFunctionsTags.EXECUTION_ARN, executionARN);
+
+        if (startDate) {
+            span.setTag(AwsStepFunctionsTags.EXECUTION_START_DATE, startDate);
+        }
+    }
+
+    private static getStateMachineName(request: any): string {
+        const stateMachineARN = get(request, 'params.stateMachineArn');
+        if (!stateMachineARN) {
+            return undefined;
+        }
+        const stateMachineARNParts = stateMachineARN.split(':');
+        return stateMachineARNParts[stateMachineARNParts.length - 1];
+    }
+
+    private static createStepFunctionTraceLink(request: any, span: ThundraSpan): void {
+        try {
+            const originalInput = get(request, 'params.input');
+
+            if (originalInput) {
+                const parsedInput = JSON.parse(originalInput);
+                const traceLink = Utils.generateId();
+
+                parsedInput[THUNDRA_TRACE_LINK_KEY] = { trace_link: traceLink, step: 0 };
+
+                span.setTag(AwsStepFunctionsTags.EXECUTION_INPUT, originalInput);
```

Comment:
I think, it is better to put this before `JSON.parse` above so even if the input is not in JSON format, we can still see the original input through tags